pkg/settlement/swap/chequebook/chequestore_test.go

```diff
@@ -0,0 +1,364 @@
+	cumulativePayout := big.NewInt(10)
+	cumulativePayoutLower := big.NewInt(5)
+	chequebookAddress := common.HexToAddress("0xeeee")
+	sig := make([]byte, 65)
+	chainID := int64(1)
+
+	chequestore := chequebook.NewChequeStore(
+		store,
+		&backendMock{},
+		&factoryMock{
+			verifyChequebook: func(ctx context.Context, address common.Address) error {
+				return nil
+			},
+		},
+		chainID,
+		beneficiary,
+		func(address common.Address, b bind.ContractBackend) (chequebook.SimpleSwapBinding, error) {
+			return &simpleSwapBindingMock{
+				issuer: func(*bind.CallOpts) (common.Address, error) {
+					return issuer, nil
+				},
+				balance: func(*bind.CallOpts) (*big.Int, error) {
+					return cumulativePayout, nil
+				},
+			}, nil
+		},
+		func(c *chequebook.SignedCheque, cid int64) (common.Address, error) {
+			return issuer, nil
+		})
+
+	_, err := chequestore.ReceiveCheque(context.Background(), &chequebook.SignedCheque{
+		Cheque: chequebook.Cheque{
+			Beneficiary:      beneficiary,
+			CumulativePayout: cumulativePayout,
+			Chequebook:       chequebookAddress,
+		},
+		Signature: sig,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = chequestore.ReceiveCheque(context.Background(), &chequebook.SignedCheque{
+		Cheque: chequebook.Cheque{
+			Beneficiary:      beneficiary,
+			CumulativePayout: cumulativePayoutLower,
+			Chequebook:       chequebookAddress,
+		},
+		Signature: sig,
+	})
+	if err == nil {
+		t.Fatal("accepted lower amount cheque")
+	}
+	if !errors.Is(err, chequebook.ErrChequeNotIncreasing) {
+		t.Fatalf("wrong error. wanted %v, got %v", chequebook.ErrChequeNotIncreasing, err)
+	}
+}
+
+func TestReceiveChequeInvalidChequebook(t *testing.T) {
+	store := storemock.NewStateStore()
+	beneficiary := common.HexToAddress("0xffff")
+	issuer := common.HexToAddress("0xbeee")
+	cumulativePayout := big.NewInt(10)
+	chequebookAddress := common.HexToAddress("0xeeee")
+	sig := make([]byte, 65)
+	chainID := int64(1)
+
+	chequestore := chequebook.NewChequeStore(
+		store,
+		&backendMock{},
+		&factoryMock{
+			verifyChequebook: func(ctx context.Context, address common.Address) error {
+				return chequebook.ErrNotDeployedByFactory
+			},
+		},
+		chainID,
+		beneficiary,
+		func(address common.Address, b bind.ContractBackend) (chequebook.SimpleSwapBinding, error) {
+			return &simpleSwapBindingMock{
+				issuer: func(*bind.CallOpts) (common.Address, error) {
+					return issuer, nil
+				},
+				balance: func(*bind.CallOpts) (*big.Int, error) {
+					return cumulativePayout, nil
+				},
+			}, nil
+		},
+		func(c *chequebook.SignedCheque, cid int64) (common.Address, error) {
+			return issuer, nil
+		})
+
+	_, err := chequestore.ReceiveCheque(context.Background(), &chequebook.SignedCheque{
+		Cheque: chequebook.Cheque{
+			Beneficiary:      beneficiary,
+			CumulativePayout: cumulativePayout,
+			Chequebook:       chequebookAddress,
+		},
+		Signature: sig,
+	})
+	if err == nil {
+		t.Fatal("accepted cheque from invalid chequebook")
+	}
```

Comment:
This check is not needed, it is covered with the next if statement.

Reply:
fixed